In [1]:
import datetime
import os
from mdagent import MDAgent
import matplotlib.pyplot as plt

In [2]:
prompt5 = "Simulate 1A3N and 7VDE (two PDB IDs matching hemoglobin) with identical parameters. Find the appropriate parameters for simulating hemoglobin from the literature. Then, plot the radius of gyration throughout both simulations."
llm_var = "gpt-4-turbo-2024-04-09"
tools = "all"
agent = MDAgent(agent_type="Structured", model=llm_var, top_k_tools=tools)
now = datetime.datetime.now()
date = now.strftime("%Y-%m-%d")
print("date and time:",date)
time = now.strftime("%H:%M:%S")
print("time:",time)
print("LLM: ",agent.llm.model_name,"\nTemperature: ",agent.llm.temperature)

date and time: 2024-09-05
time: 16:09:03
LLM:  gpt-4-turbo-2024-04-09 
Temperature:  0.1


In [3]:
agent.run(prompt5)

Thought: To address the question, I need to first find appropriate simulation parameters for hemoglobin from the literature. Then, I will set up and run molecular dynamics simulations for both PDB IDs (1A3N and 7VDE) using these parameters. Finally, I will plot the radius of gyration throughout both simulations.

Action:
```json
{
    "action": "LiteratureSearch",
    "action_input": {
        "query": "appropriate molecular dynamics simulation parameters for hemoglobin"
    }
}
```"molecular dynamics simulation parameters hemoglobin"
Search: "molecular dynamics simulation parameters hemoglobin"

Found 11 papers
Now that I have the appropriate simulation parameters for hemoglobin, I need to set up and run molecular dynamics simulations for both PDB IDs (1A3N and 7VDE) using these parameters. I will use the SetUpandRunFunction tool to perform these simulations.

Action:
```json
{
    "action": "SetUpandRunFunction",
    "action_input": {
        "pdb_id": "1A3N",
        "forcefield_fil

({'input': '\n    You are an expert molecular dynamics scientist, and\n    your task is to respond to the question or\n    solve the problem to the best of your ability using\n    the provided tools.\n\n    You can only respond with a single complete\n    \'Thought, Action, Action Input\' format\n    OR a single \'Final Answer\' format.\n\n    Complete format:\n    Thought: (reflect on your progress and decide what to do next)\n    Action:\n    ```\n    {\n        "action": (the action name, it should be the name of a tool),\n        "action_input": (the input string for the action)\n    }\n    \'\'\'\n\n    OR\n\n    Final Answer: (the final response to the original input\n    question, once all steps are complete)\n\n    You are required to use the tools provided,\n    using the most specific tool\n    available for each action.\n    Your final answer should contain all information\n    necessary to answer the question and its subquestions.\n    Before you finish, reflect on your pro

In [4]:
registry = agent.path_registry
print(registry.list_path_names_and_descriptions())

No names found. The JSON file is empty or does not contain name mappings.


In [5]:
now = datetime.datetime.now()
date = now.strftime("%Y-%m-%d")
print("date and time:",date)
time = now.strftime("%H:%M:%S")
print("time:",time)

date and time: 2024-09-05
time: 16:14:24


In [ ]:
paths_and_descriptions = registry.list_path_names_and_descriptions()
print("\n".join(paths_and_descriptions.split(",")))

In [ ]:
# SKIPPED below --- agent fails to call tools

In [ ]:
# from IPython.display import Image
# path_1 = registry.get_mapped_path("fig0_175712")
# path_2 = registry.get_mapped_path("fig0_175847")

In [ ]:
# Image(path_1)

In [ ]:
# Image(path_2)

# Experiment Result:
### Completed without Exception or TimeOut Errors ❌
### Attempted all necessary steps ❌
### No hallucination ✅
### Logic make sense ❌
### Correct Answer  ❌
### Number of steps completed: 1
### % steps completed: 14%